### PR TITLE
WAF-38: Get image urls from product search query instead of refined product query for each item

### PR DESCRIPTION
**Pull request by yugandhar02**
This PR fixes 
1. Performance issue by removed refined product query call and getting images from product search query based on new "eds_images" attribute
2. Show Add to bag button only for sportswear category
3. Fix columns for various viewports

### CODE REVIEW

File: src/utils/getProductImage.ts [hunk 1]
```diff
@@ -107,4 +108,50 @@ const generateOptimizedImages = (
   return imageUrlArray;
 };
 
-export { generateOptimizedImages, getProductImageURLs };
+function getProductImagesFromQuery(item: Product) {
```
Comment (chicharr) [inline src/utils/getProductImage.ts:111]: Not a blocker: but we should probably change the name to `getProductImagesFromAttribute`, the current name sounds like you are running a new query

File: src/utils/getProductImage.ts [hunk 2]
```diff
@@ -107,4 +108,50 @@ const generateOptimizedImages = (
...
+  const { productView } = item;
+  const attributeId = productView?.options?.[0].id;
+  if (!attributeId) {
+    return [];
+  }
+
+  const imageAttributes = productView?.attributes?.find(({ name }) => name === 'eds_images');
```
Comment (chicharr) [inline src/utils/getProductImage.ts:118]: Would be great if you can add a comment explaining all this crazy logic to select the right image.

File: src/utils/productUtils.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+function getColorSwatchesFromQuery(item: Product) {
+  if (isSportsWear(item)) {
+    return [];
+  }
+
+  const { productView } = item;
+  const attributeId = productView?.options?.[0].id;
+  const imageAttributes = productView?.attributes?.find(({ name }) => name === 'eds_images');
+  if (imageAttributes) {
```
Comment (chicharr) [inline src/utils/productUtils.ts:18]: same as before, the requirements to select the images are quite confusing, so would be great if you can explain them with comments in the code

**Review by chicharr — APPROVED** [review]

File: src/utils/getProductImage.ts (hunk 1 above)
Comment (yugandhar02) [reply src/utils/getProductImage.ts]: Updated the function name

File: src/utils/getProductImage.ts (hunk 2 above)
Comment (yugandhar02) [reply src/utils/getProductImage.ts:130]: Added comment

**Pull request merged**